src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -758,10 +758,23 @@ void executeSuppressionActions(IEnumerable<Diagnostic> reportedDiagnostics, bool
                 var suppressors = this.Analyzers.OfType<DiagnosticSuppressor>();
                 if (concurrent)
                 {
-                    Parallel.ForEach(suppressors, suppressor =>
```

Comment:
>Parallel.ForEach(suppressors, suppressor => [](start = 20, length = 43)

The motivation for this change is not obvious. Could you please share some details?

Reply:
`OperationCanceledException` during the suppressor execution leads to it being wrapped in an `AggregateException` by `Parallel.ForEach`, which ends up being reported as an NFW by our exception handlers, see https://github.com/dotnet/roslyn/issues/41713#issuecomment-588505199.

This change switches to the task based approach used at other places in the driver, which also avoids this case. I can add a comment.

Reply:
> This change switches to the task based approach used at other places in the driver, which also avoids this case. I can add a comment.

So, despite what the title of this PR says, this change prevents wrapping of an OperationCanceledException with AggregateException rather than does any kind of unwrapping. Please make sure this is reflected in the commit comment on merge.


---
In reply to: [384078808](https://github.com/dotnet/roslyn/pull/41892#discussion_r384078808) [](ancestors = 384078808)

Reply:
Ah yes, that was my previous attempt. Let me update the PR title. I'll also update the commit comment prior to merge.